sig-scalability/charter.md

```diff
@@ -0,0 +1,170 @@
+# SIG Scalability Charter
+
+## Mission
+The SIG Scalability helps to define scalability goals for Kubernetes, ensures
+that they all play well together and ensures that every Kubernetes release meets
+them by measuring performance and scalability indicators and publishing the
+results.
+
+We also coordinate and contribute to general system-wide scalability and
+performance improvements (that don’t fall into the charter of another individual
+SIG) as well as provide consultations about any scalability and performance
+related aspects of Kubernetes.
+
+## What can we do/require from other SIGs
+Scalability and performance are horizontal aspects of the system - changes in a
+single place of Kubernetes may affect the whole system. As a result, to
+effectively ensure Kubernetes scales, we need a special cross-SIG privileges.
+
+- We can rollback any merged PR if it has been identified as a cause of any
+  [performance/scalability SLOs] regression. The offending PR should only be
+  merged again after proving to pass  tests at scale.
```

Comment:
Is that an automated process? e.g. as a PR owner do I know if my code has violated SLOs?

Reply:
Partially. So we already have some presubmits that are checking that on smaller scale.
And for larger scale (i.e. 5k nodes) it's impossible to have presubmits because they are way too long and way too expensive.
So that is being checked on postsubmits and as of now monitores by sig-scalability. We are working on introducing more automation here, but that's not yet done.

Reply:
We don't necessarily have to do this within this charter, but I'm going to suggest that the release team require regular participation from a member of SIG Scalability until we have such automation in place.  Chasing down scalability related failures is too much an additional burden on the CI Signal role, and we often have little visibility into whether or how failures are being worked.

As an example both https://k8s-testgrid.appspot.com/sig-scalability-gce#gce-scale-performance and https://k8s-testgrid.appspot.com/sig-scalability-gce#gce-scale-correctness have been failing for over two weeks.  This is being "explained away" as a quota issue, with code continuing to merge, rather than being enforced as a hard stop until the cause is identified.

Reply:
It's definitely not your role (as sig release) to track the failures - I agree.

But again, whether we should be on sig release meetings or not, is an implementation detail. We are responsible for those tests and it's possible that this means that we need to come to sig release.
But that's implementation detail of the mission, not part of the charter.